### ISSUE TITLE
Is this a programming joke or not?

### ISSUE THREAD

**Issue by 7heo** [drathier/stack-overflow-import#11]
A colleague and I can't decide if this is a programming joke or if it is meant seriously. Could you please shed some light there? :)

PS: This isn't a meta-joke, it is a serious question.

**Comment by osilkin98** [comment]
no

**Comment by gjask** [comment]
Since our company uses `stack-overflow-import` as usual solution to arbitrary algorithmic problems our productivity skyrocketed. We use it for serious business as it seems quite production ready even for enterprise tier.

**Comment by tcpj** [comment]
Your company too? It's pretty common I see.

**Comment by 7heo** [comment]
> **osilkin98** commented 3 hours ago
> no

I think we should wait for @drathier to actually answer. Anyone can guess, but nobody can actually know the original intent.

**Comment by refi64** [comment]
Well, based on the licensing statement:

> This module is licensed under whatever license you want it to be

and [these comments](https://github.com/drathier/stack-overflow-import/pull/2#issuecomment-258418150), I'm inclined to believe it's satire...

**Comment by 7heo** [comment]
@kirbyfan64 yeah I knew about the license, it's what my colleague and I didn't agree on. It might also be that @drathier is annoyed with licenses and legal speak.

On the other hand, the comments are new to me, and I would agree that they seem to indicate satire. However, I'm not convinced yet.

**Comment by gguliani2U** [comment]
Please be sure to test if this works for you during leap years and months with solar eclipses!

**Comment by yashovardhan99** [comment]
Wow! Issue has been open for a year and we still don't know if this is real or not. Anyone who still follows this?

**Comment by amcgregor** [comment]
Only the original author can answer the question at this point, however I often guide people this way as a satirical example implementation of a broader-known joke concept.  Where a joke could be taken seriously; there exist things that can become [beautiful in their ugliness](https://www.youtube.com/playlist?list=PL7X2jUZPmG7JIOYnODSINRzl1PdmhzLiZ).  (Up to 13 or 14 viewings of Manos… content warnings on some of those, but they're mostly just wildly adult WTF, **enjoy Darkplace if you haven't heard of it!**)  This being a true implementation of [this parody of the O'Reilly Media Inc. series famous for their topic-specific theme animals](https://www.goodreads.com/book/show/29437996-copying-and-pasting-from-stack-overflow). This library is genius.  Evil genius, _but still genius_.

To dive a touch deeper, that topical/titling parody may be similarly inspired by [this one targeting the use of animals](https://f.v1.n0.cdn.getcloudapp.com/items/2B171b24450k122Y1S47/_S0BtP.jpg) — the shocked expression of absolute terror is golden. A ["fun pastime with co-workers over beer"](https://ghcdn.webcore.io/oreally-wtf-series/classic.jpeg) to which I have made an [occasional contribution](https://ghcdn.webcore.io/oreally-wtf-series/git.png) via my "What the F*" mockuseries.

Ultimately: if it works, does it matter?  And regardless of the safety or reliability in using this, it can be used safely and reliably, if done so carefully.  E.g. review the code it's executing and freeze that code so it isn't entirely dynamic on every invocation, as minimal security and reliability precautions. [AL/№ 57](https://gist.github.com/amcgregor/9f5c0e7b30b7fc042d81#file-_laws-md): If something is worth doing twice, it’s worth building a tool to do it.

Note, however, your code will fail various certifications and audits if graced by the presence of code directly attributable back to StackOverflow, in a number of situations.  Such as a recent independent third party source-level audit my project underwent prior to acquisition, a few years back.  Dinged for one line, went back to the answer, looked him up on GitHub, lo, he had packaged up that answer as a pip-installable solution with acceptable license.  🧈🥬💃

**Comment by matthewdeanmartin** [comment]
I independently thought of the same joke and wrote way too much code to actually turn SO answers into python packages suitable for vendorizing.

https://github.com/matthewdeanmartin/so_pip

There are many problems with taking the joke seriously, among them is the really small % of answers that are in the form of re-usable code. That said, a handful of answer are good candidates for "abusing" SO as a form of source control.
